src/core/defs.jl

```diff
@@ -195,12 +220,13 @@ Set the values of `md` dimension `name` to integers 1 through `count`, if `keys`
 an integer; or to the values in the vector or range if `keys` is either of those types.
 """
 function set_dimension!(md::ModelDef, name::Symbol, keys::Union{Int, Vector, Tuple, Range})    
+    if name == :time
+        reset_run_periods!(md, keys[1], keys[end])
+        md.is_uniform = isuniform(keys)
+    end
     if haskey(md, name)
         warn("Redefining dimension :$name")
     end
```

Comment:
@rjplevin I tried to simplify this one like you recommended by having name==:time just once, however unless I want to have another separate call to haskey() within this conditional, it will call reset_run_periods! even if the model doesn't already have a time index. I think this is okay, though, because if it doesn't already have a time index, then it doesn't have any components, so reset_run_periods! will loop over an empty dictionary. It still seems a bit inelegant though..

Reply:
How about:
```
function set_dimension!(md::ModelDef, name::Symbol, keys::Union{Int, Vector, Tuple, Range})
    redefined = haskey(md, name)
    if redefined
        warn("Redefining dimension :$name")
    end

    if name == :time
        md.is_uniform = isuniform(keys)
        if redefined:
            reset_run_periods!(md, keys[1], keys[end])
    end
    
    dim = Dimension(keys)
    md.dimensions[name] = dim
    return dim
end
```

Reply:
ah so clever, thank you!

Reply:
Whoops... some Python syntax snuck in my code.